sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -152,8 +152,9 @@ case class HashAggregateExec(
   override def usedInputs: AttributeSet = inputSet
 
   override def supportCodegen: Boolean = {
-    // ImperativeAggregate is not supported right now
-    !aggregateExpressions.exists(_.aggregateFunction.isInstanceOf[ImperativeAggregate])
+    // ImperativeAggregate and filter predicate are not supported right now
```

Comment:
Filed: https://issues.apache.org/jira/browse/SPARK-30027
Can you leave some comments about this here like `TODO: SPARK-30027 Support codegen for filter exprs in HashAggregateExec`?

Reply:
OK